sdcm/tester.py

```diff
@@ -520,6 +529,143 @@ def _setup_ldap_roles(self, db_cluster: BaseScyllaCluster):
+        unique_member_update = {'uniqueMember': [
+            ('MODIFY_DELETE', [f'uid={member_name},ou=Person,{LDAP_BASE_OBJECT}'])]}
+        res = self.localhost.modify_ldap_entry(distinguished_name, unique_member_update)
+        if not res and raise_error:
+            raise Exception(f'Failed to delete entry {distinguished_name} from Ldap role {ldap_role_name}')
+
+    def delete_ldap_role(self, ldap_role_name: str, raise_error: bool = True):
+        distinguished_name = self.search_ldap_role(ldap_role_name=ldap_role_name, raise_error=raise_error)
+        self.log.debug("Deleting Ldap entry: %s", distinguished_name)
+        res = self.localhost.delete_ldap_entry(distinguished_name)
+        if not res and raise_error:
+            raise Exception(f'Failed to delete entry {distinguished_name} from Ldap.')
+
+    @retrying(n=10, sleep_time=30, message='Waiting for no user permissions verification', allowed_exceptions=(ValueError,))
+    def wait_verify_user_no_permissions(self, username: str):
+        """
+        Checks for unauthorized user permission following roles update.
+        Creating a keyspace by the user and verify it fails as expected.
+        """
+        self.log.debug("Verifying user %s roles permission change in Scylla DB (following an LDAP Role update) ", username)
+
+        try:
+            with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0], user=username,
+                                                        password=LDAP_PASSWORD) as session:
+                session.execute(
+                    """ CREATE KEYSPACE IF NOT EXISTS test_no_permission WITH replication = {'class': 'SimpleStrategy',
+                    'replication_factor': 1} """)
+                session.execute(""" DROP KEYSPACE IF EXISTS test_no_permission """)
+            raise ValueError('DID NOT RAISE')
+        except Unauthorized:
+            return
+
+    @retrying(n=10, sleep_time=30, message='Waiting for user permissions update', allowed_exceptions=Unauthorized)
+    def wait_verify_user_permissions(self, username: str):
+        """
+        Checks for authorized user permission following roles update.
+        Creating a keyspace by the user and verify it doesn't fail.
+        """
+        self.log.debug("Verifying user %s roles permission change in Scylla DB (following an LDAP Role update) ", username)
+
+        with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0], user=username,
+                                                    password=LDAP_PASSWORD) as session:
+            session.execute(
+                """ CREATE KEYSPACE IF NOT EXISTS test_permission_ks WITH replication = {'class': 'SimpleStrategy',
+                'replication_factor': 1} """)
+            session.execute(""" DROP KEYSPACE IF EXISTS test_permission_ks """)
+
+            session.execute(
+                """ CREATE KEYSPACE IF NOT EXISTS customer WITH replication = {'class': 'SimpleStrategy',
```

Comment:
don't we use `customer` as keyspace name in some of the user profiles we have? i suggest changing it to something like `customer_ldap` to be 100% sure it will be unique

Reply:
`customer` name is not in use elsewhere

Reply:
fixeing to customer_ldap anyway